src/components/TextInput/BaseTextInput.js

```diff
@@ -94,6 +102,12 @@ class BaseTextInput extends Component {
      * @memberof BaseTextInput
      */
     setValue(value) {
+        if (this.props.exposeValueToRef) {
+            this.props.exposeValueToRef(this.input, value);
+        }
```

Comment:
Could we also solve this by always setting ref's value no matter what instead of using a callback? 
The behavior should be work the same on native and web?

`this.props.exposeValueToRef` in a way is saying "only do this on native" but if web + native can work the same what is to stop us from using the same method on both platforms skip relying on `e.target.value` in web entirely?

Reply:
Is there some way to make `ref.value` always refer to `this.value` without explicitly setting it anytime `this.value` is modified?

Thinking it might be better because we should potentially update it here as well...

https://github.com/Expensify/App/blob/73a44eef223cda61488bd483e171c6831c47b326/src/components/TextInput/BaseTextInput.js#L57-L63

Reply:
After looking at this `TextInput` component I'm wondering if we should also address this in a follow-up. But I do have a concern that the next component we create will follow similar patterns. e.g. the next component someone goes to create will they implement a `exposeValueToRef` or should they try something else?

Here's where I think this particular `TextInput` component went wrong...

1. We are updating `this.value` instead of a shared `this.ref.value`
2. We should pass an internal ref to `props.innerRef()` and not pass the child ref (which has no ability to access `value` on native)
3. This way whenever we update `this.value` (now, `this.input.value`) the component itself can reference that value, but also the parent can reference it without explicitly updating both an internal value _and_ calling a function to decorate the ref with that value.

Maybe we can come up with some kind of utility to make this easier.

Reply:
I'm not too sure that I understand how this suggestion works. I'm happy to talk over it to try to understand it though.

As of now, the best I've got is to set `this.input.value = this.value` in the `render()` and `setValue` methods of `TextInput`, that way it's consistent across platforms and we have access to the value in the ref.

Reply:
Ok, let's punt it for now. I'm not sure I'll have time to look into it and can't think of the exact solution without doing a deep dive.

Basically my concern is that we are treating web and native slightly differently WRT when their values are available. Another alternative to develop that consistency would be to store a separate object of values instead of using `this.inputRef.value` directly.

So we'd have 

```js
this.inputValues = {};
```
And then populate it with default values + modify them in an input's `onChange` and never refer directly to a ref's `value` property since the behavior on web + mobile is different. Does this sound OK?

Reply:
Sorry, just saw this after I pushed my changes. That sounds good and I'll work on it next. Thanks!

Reply:
I updated it with the new `this.inputValues` object